### PR TITLE
docs: include deprecation plan and migration guide for firebase_admob to google_mobile_ads

### PR DESCRIPTION
**Pull request by bparrishMines**
## Description

Include deprecation plan and migration guide in README.

Readme in markdown: https://github.com/bparrishMines/flutterfire/blob/deprecation/packages/firebase_admob/README.md

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/firebase_admob/README.md [hunk 1]
```diff
@@ -1,3 +1,171 @@
+# Deprecation of firebase_admob Plugin
+The `firebase_admob` plugin will be deprecated in April 2021 in favor of [Google Mobile Ads SDK for
```
Comment (blasten) [inline packages/firebase_admob/README.md:2]: nit: add line

File: packages/firebase_admob/README.md [hunk 2]
```diff
@@ -1,3 +1,171 @@
...
+       anchorType: AnchorType.bottom,
+     );
+ }
+
+ @override
+ Widget build(BuildContext context) {
+   return MyAppWidget();
+ }
+}
+```
```
Comment (blasten) [inline packages/firebase_admob/README.md:63]: Indentation is somewhat off. single space used for tabs in some places here and below

File: packages/firebase_admob/README.md [hunk 3]
```diff
@@ -1,3 +1,171 @@
...
+ @override
+ void initState() {
+   super.initState();
+   _interstitialAd = InterstitialAd(
+     adUnitId: InterstitialAd.testAdUnitId,
+     request: AdRequest(),
+     listener: AdListener(
+       onAdLoaded: (Ad ad) {
+         // Ad is now ready to show at any time.
+         // (ad as InterstitialAd).show();
```
Comment (blasten) [inline packages/firebase_admob/README.md:143]: should it call `show`, or leave it as an option to the developer?

Comment (bparrishMines) [reply packages/firebase_admob/README.md:143]: It's bad practice to call `show` here. A user is supposed to preload an Ad before its needed to be shown. I'll just remove this line and leave the comment above.

**Review by blasten — APPROVED** [review]
LGTM

**Review by kroikie — APPROVED** [review]

**Comment by Ehesp** [comment]
Thanks all!

**Pull request merged**